UI/FleetWnd.cpp

```diff
@@ -3532,29 +3528,18 @@ void FleetWnd::FleetRightClicked(GG::ListBox::iterator it, const GG::Pt& pt, con
     {
         auto split_action = [this, &ship_ids_set]() {
             ScopedTimer split_fleet_timer("FleetWnd::SplitFleet", true);
-            // remove first ship from set, so it stays in its existing fleet
-            auto ship_id_it = ship_ids_set.begin();
-            ship_ids_set.erase(ship_id_it);
```

Comment:
This code mean that the oldest ship in the fleet always kept the old aggression value. I don't see why it should.

Reply:
As far as I could make out, the oldest ship is always the first, but that is not the point.
There is nothing special about a fleet, they are just collections of ships with number.
Keeping the first ship in the fleet means that they player may end up with an aggressive scout just because this scout happened to be the first ship in its fleet.

Reply:
I think in this case, when splitting a fleet in a way that treats all ships the same, it's reasonable that this action would effectively (or really) make new fleets for each ship (or each design) and thus not treat the one ship that remains in the original fleet differently from the others. It can set the aggression of the resulting fleets independently, based on their design, if the UI is set to auto-determine fleet aggression, but if it's set to a particular aggression for new fleets, then they should all get that aggression, not all but the original fleet.